api_app/analyzers_manager/observable_analyzers/joesandbox.py

```diff
@@ -0,0 +1,194 @@
+import logging
+
+from jbxapi import JoeSandbox
+
+from api_app.analyzers_manager.classes import ObservableAnalyzer
+from api_app.analyzers_manager.exceptions import AnalyzerRunException
+from api_app.mixins import JoeSandboxMixin
+from tests.mock_utils import if_mock_connections, patch
+
+logger = logging.getLogger(__name__)
+
+
+class JoeSandboxAnalyzer(ObservableAnalyzer, JoeSandboxMixin):
+    url: str
+    sample_at_url: bool = False
+
+    def submit_observable(self, sandbox_session: JoeSandbox) -> str:
+        logger.info(f"Submitting observable: {self.observable_name}")
+        params = {"systems": self.system_to_use}
+        submission: dict = (
+            sandbox_session.submit_sample_url(self.observable_name, params=params)
+            if self.sample_at_url
+            else sandbox_session.submit_url(self.observable_name, params=params)
+        )
+
+        logger.info(
+            f"Observable submitted successfully with submission id: {submission['submission_id']}"
+        )
+        return submission["submission_id"]
+
+    @classmethod
+    def update(cls):
+        pass
+
+    def run(self):
+        sandbox_session = JoeSandbox(
+            apikey=self._api_key, apiurl=self.url, accept_tac=True
+        )
+
+        # checking if similar submission in account is already present
+        analysis_id = self.check_submission_exists(
+            session=sandbox_session, observable_url=self.observable_name
+        )
+        if analysis_id:
+            return {analysis_id: sandbox_session.analysis_info(analysis_id)}
+
+        logger.info("Existing submission not found")
+
+        # checking if similar analysis is present in public DB
+        logger.info(f"Checking if analysis is present for {self.observable_name}")
+
+        analysis_ids = self.check_if_analysis_present(
```

Comment:
good idea these 2 checks, still this would be better to be configurable with an optional parameter in the analyzer. Something like "force_new_analysis" that defaults to False but if True would skip this checks and directly create a new submission. Sometimes the user would like to just create a new one instead of getting the old one